jdaviz/configs/default/plugins/collapse/collapse.py

```diff
@@ -74,30 +62,28 @@ def _on_data_updated(self, msg):
 
     @observe('selected_data_item')
     def _on_data_item_selected(self, event):
-        self._selected_data = next((x for x in self.data_collection
-                                    if x.label == event['new']))
+        data_label = event['new']
+        if data_label not in self.data_collection.labels:
+            return
+        self._selected_data = self.data_collection[self.data_collection.labels.index(data_label)]
+        self._selected_cube = self._selected_data.get_object(cls=Spectrum1D, statistic=None)
```

Comment:
Note: I feel like there is no point to extract the cube multiple times across the plugin. Here, I change it to extracting it only once when data is selected.